osf/models/user.py

```diff
@@ -1261,6 +1266,15 @@ def change_password(self, raw_old_password, raw_new_password, raw_confirm_passwo
         if issues:
             raise ChangePasswordError(issues)
         self.set_password(raw_new_password)
+        self.reset_old_password_invalid_attempts()
+
+    def increment_old_password_invalid_attempts(self):
+        self.old_password_invalid_attempts += 1
+        self.save()
```

Comment:
In general, we want to avoid implicit saves because it usually results in unnecessary database work. Can we do without these?